rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -794,3 +798,22 @@ def _is_non_conformant_erc721(address: ChecksumEvmAddress) -> bool:
     def _address_is_exchange(address: ChecksumEvmAddress) -> Optional[str]:
         """Takes an address and returns if it's an exchange in the given chain
         and the counterparty to use if it is."""
+
+    def _check_correct_types(self, decoded_events: list['EvmEvent']) -> bool:
```

Comment:
Maybe put the entire method definition under `__debug__` ?

So something like
```python
if __debug__:
    def _check_correct_types(self, decoded_events: list['EvmEvent']) -> bool:
        unexpected_types = set()
        for event in decoded_events:
            if event.counterparty is None:
                continue
```

Also afaik `if __debug__` should be used without `is ...` in order for the code to be removed in production (or at least we use it like this everywhere).

Reply:
I didn't know the debug rule thing, thanks! The function call should be removed since it has an assert in front and loading the function in the class I believe is not such a big deal. Having the `if __debug__ return` thing is just to avoid calling it without noticing in prod